.travis.yml

```diff
@@ -5,8 +5,6 @@ python: 3.7
 sudo: required
 fast_finish: true
 
-if: tag IS blank
```

Comment:
If we remove this, I think that Travis will run the tests twice when we push a new tagged commit (one build for the updated branch, plus one build for the tag). As `s2p` tests are quite slow this is a bit annoying. Do you think that we should move this condition in each of the `stage: test` jobs? Do you see a better way to avoid duplicated builds?

Reply:
After discussion, the 2 options we have are:
1. Skip the `test` stages if the build is for a tag.
    - Upside: no work required by the maintainer.
    - Downside: No guarantee that the tests pass before deploying to PyPI, so if some bugs have been introduced by committing directly to `master` and tagging (unlikely), they will not be detected

2. Leave the `.travis.yml` as is, and find a way to cancel the `master` build when we push both a `tag` and a `commit` on `master`.
    - Upside: tests are run before deploying, so we are safe.
    - Downside: some manual action required from the maintainer if no automatic way to do this is found.

Reply:
After reading this [travis issue](https://travis-ci.community/t/tagged-commit-triggers-two-builds/2975/5) I vote for option 2. I haven't found an automatic way to cancel the `master` build when we push both a tag and a commit on `master`, but it's not that big of a deal to cancel it manually or even to have a duplicated build for the releases.